src/main/kotlin/org/jetbrains/research/testspark/display/generatedTests/TopButtonsPanelBuilder.kt

```diff
@@ -41,13 +41,21 @@ class TopButtonsPanelBuilder {
      */
     fun update(generatedTestsTabData: GeneratedTestsTabData) {
         var numberOfPassedTests = 0
+        var numberOfRemovedTests = 0
         for (testCasePanelFactory in generatedTestsTabData.testCasePanelFactories) {
-            if (testCasePanelFactory.isRemoved()) continue
+            if (testCasePanelFactory.isRemoved()) {
+                numberOfRemovedTests++
+                continue
+            }
             val error = testCasePanelFactory.getError()
             if ((error is String) && error.isEmpty()) {
                 numberOfPassedTests++
             }
         }
+        if (generatedTestsTabData.testCasePanelFactories.size == numberOfRemovedTests) {
```

Comment:
The code will be easier to read if we use stream API here:
```kotlin
// ATTENTION: I did not test these expressions; only re-wrote the existing code
val passedTestsCount = generatedTestsTabData.testCasePanelFactories
            .filter { !it.isRemoved() }
            .count { it.getError()?.isEmpty() == true }
            
val removedTestsCount = generatedTestsTabData.testCasePanelFactories.count { it.isRemoved() }
``` 
Working with immutable variables makes coding reading easier here.